components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -2142,6 +2196,63 @@ std::string AdsServiceImpl::LoadResourceForId(const std::string& id) {
   return LoadDataResourceAndDecompressIfNeeded(resource_id);
 }
 
+void AdsServiceImpl::ClearScheduledCaptcha() {
```

Comment:
why do we have these same methods on AdsServiceImpl? This is really vastly more complicated than it needs to be and a perfect opportunity to create a new keyed service with a separate mojo interface using BraveWallet code as an example

Reply:
Reworked to make `BraveAdaptiveCaptchaService` a keyed service.